### PR TITLE
Fix git alias conflict with github-cli

### LINKED ISSUE #1893: Git alias takes precedence over GitHub CLI of the same name
## Expected Behavior

When git aliases are enabled, the `gh` alias does _not_ take precedence over the [GitHub CLI](https://cli.github.com/), which is invoked by the same name.

## Current Behavior

`gh` currently invokes the git alias, not the GitHub CLI.
https://github.com/Bash-it/bash-it/blob/e06066c36768ac08a31927646a150a6e66badbfc/aliases/available/git.aliases.bash#L89-L91

## Possible Solution

A few possibilities that might have caused the bug
1. The `_command_exists` function
2. Whether git aliases are enabled _before_ or _after_ the GitHub CLI has been installed. In my case, I enabled the aliases before installing the CLI. Perhaps if I flip the order, the bug will go away

## Context

In my view, when an alias happens to have the exact same name as a command line program, priority should be given to the program. This goes beyond just the GitHub CLI.

## Steps to Reproduce

1. `bash-it enable alias git`
2. `brew install gh`
3. `gh --help`: Instead of seeing the help message for the GitHub CLI, I see nothing

## Your Environment

<details><summary>Bash-it version used</summary>

```
Version type: dev
Current git SHA: 6093ecb on 2021-02-26T10:17:01+02:00
Commit info: https://github.com/Bash-it/bash-it/commit/6093ecb
Compare to latest: https://github.com/Bash-it/bash-it/compare/6093ecb...master
```

</details>

<details><summary>List of enabled plugins, themes and aliases</summary>

- Plugins: `alias-completion`, `base`
- Themes: `bobby-python`
- Aliases: `git`

</details>

<details><summary>bash-it doctor</summary>

```
DEBUG: core: main: Loading libraries(except appearance)...
DEBUG: lib: helpers: Loading library file...
DEBUG: lib: log: Loading library file...
DEBUG: lib: preview: Loading library file...
DEBUG: lib: search: Loading library file...
DEBUG: lib: utilities: Loading library file...
DEBUG: vendor: Loading "preexec"...
DEBUG: core: reloader: Loading all enabled components...
DEBUG: aliases: git: Loading component...
DEBUG: plugin: base: Loading component...
DEBUG: completion: bash-it: Loading component...
DEBUG: completion: system: Loading component...
DEBUG: plugin: alias-completion: Loading component...
DEBUG: core: reloader: Loading "bobby-python" theme...
 WARN: themes: base: Command 'battery_percentage' does not exist!
DEBUG: core: main: Loading custom aliases, completion, plugins...
DEBUG: core: main: Loading general custom files...
DEBUG: custom: example: Loading custom file...
```

</details>

<details><summary>Bash version</summary>

GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19)

</details>

<details><summary>Operating System and version</summary>

macOS Catalina 10.15.7

</details>

## Your Bash Config File

<details><summary>.bash_profile</summary>

```bash
#!/usr/bin/env bash

# If not running interactively, don't do anything
case $- in
  *i*) ;;
    *) return;;
esac

export BASH_IT="/Users/zhengnan/.bash_it"
export BASH_IT_THEME='bobby-python'

# Your place for hosting Git repos. I use this for private repos.
export GIT_HOSTING='git@git.domain.com'

# Don't check mail when opening terminal.
unset MAILCHECK

# Change this to your console based IRC client of choice.
export IRC_CLIENT='irssi'

# Set this to the command you use for todo.txt-cli
export TODO="t"

# Set this to false to turn off version control status checking within the prompt for all themes
export SCM_CHECK=true

# Set Xterm/screen/Tmux title with only a short hostname.
# Uncomment this (or set SHORT_HOSTNAME to something else),
# Will otherwise fall back on $HOSTNAME.
#export SHORT_HOSTNAME=$(hostname -s)

# Set Xterm/screen/Tmux title with only a short username.
# Uncomment this (or set SHORT_USER to something else),
# Will otherwise fall back on $USER.
#export SHORT_USER=${USER:0:8}

# Set Xterm/screen/Tmux title with shortened command and directory.
# Uncomment this to set.
#export SHORT_TERM_LINE=true

# (Advanced): Uncomment this to make Bash-it reload itself automatically
# after enabling or disabling aliases, plugins, and completions.
# export BASH_IT_AUTOMATIC_RELOAD_AFTER_CONFIG_CHANGE=1

# Uncomment this to make Bash-it create alias reload.
# export BASH_IT_RELOAD_LEGACY=1

# Load Bash It
source "$BASH_IT"/bash_it.sh
```

</details>

### PR DESCRIPTION
**Pull request by tbhaxor**
## Description

Fixed type `_command_exists` by replacing it with `_binary_exists`

## Motivation and Context

Fixes #1893

## How Has This Been Tested?

Tested on local system with sourcing

## Screenshots (if appropriate):

**Before**

Screenshot_20210613_045951

**After**

Screenshot_20210613_050022

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Review by NoahGorny — COMMENTED** [review]
Hi @tbhaxor, thanks for taking the effort in trying to solve this!
I do not think this actually fixes the problem in #1893, but I am considering merging this anyway.
The thing is, you are probably correct we should check for the binary, but the current syntax is also correct, if `gh` is already aliases/defined, maybe we do not want to re-alias it

**Comment by tbhaxor** [comment]
> The thing is, you are probably correct we should check for the binary, but the current syntax is also correct, if `gh` is already aliases/defined, maybe we do not want to re-alias it

@NoahGorny Remember once I had a problem with github-cli binary being overridden by alias.

https://github.com/Bash-it/bash-it/issues/1774

Maybe we have missed out (intentionally or due to workload)  this alias

**Comment by tbhaxor** [comment]
@NoahGorny Ping

File: aliases/available/git.aliases.bash [hunk 1]
```diff
@@ -86,7 +86,7 @@ alias ggui='git gui'
 # home
 alias ghm='cd "$(git rev-parse --show-toplevel)"' # Git home
 # appendage to ghm
-if ! _command_exists gh; then
+if ! _binary_exists gh; then
```
Comment (davidpfarrell) [inline aliases/available/git.aliases.bash:89]: If the user creates their own `gh` alias, its not our place to override it ... `_command_exists` seems correct here to me.

Comment (tbhaxor) [reply aliases/available/git.aliases.bash:89]: But we shouldn't override it with the GitHub CLI binary. Should we?

**Comment by davidpfarrell** [comment]
Seems like the real question in your setup is:

How is `_command_exists gh` failing at this point in the bashit startup, since it _appears_ that it should **always** be present in your `/usr/bin` ?

Your `"Before"` section didn't include a `type bh` _before_ sourcing `git.aliasse.bash`

**Comment by tbhaxor** [comment]
I got your point now after running this example in the terminal. `_command_exists` is the right thing here

**Pull request closed without merging**